hack/examples/bases/nutanix/clusterclass/kustomization.yaml.tmpl

```diff
@@ -46,16 +46,23 @@ patches:
     - op: "remove"
       path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/certSANs"
 
-# Delete the kube-vip file.
-# Will be templated and added back in the handler if enabled.
+# Template the kube-vip file.
+# The handler will set the variables if needed, or remove it.
 - target:
     kind: KubeadmControlPlaneTemplate
   patch: |-
     - op: test
       path: /spec/template/spec/kubeadmConfigSpec/files/0/path
       value: "/etc/kubernetes/manifests/kube-vip.yaml"
     - op: "remove"
-      path: "/spec/template/spec/kubeadmConfigSpec/files/0"
+      path: "/spec/template/spec/kubeadmConfigSpec/files/0/owner"
```

Comment:
why are we removing the owner. should we set it to `root:root` explicitly or it will be defaulted to it?
also we should be setting file permissions `0600` too. https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/main/pkg/handlers/generic/mutation/controlplanevirtualip/providers/kubevip.go#L22-L24

Reply:
We only need the templated Pod spec, all other fields here are set by the handler.
I removed it intentionally so that we don't have unused, (and potentially different information) in this KCPTemplate.

Reply:
👍 I noticed it in handler code (hidden in github diff view). Also the unit tests are asserting this. Thanks.